testing/src/main/java/io/grpc/internal/testing/AbstractTransportTest.java

```diff
@@ -177,6 +178,22 @@ public void setUp() {
     when(serverStreamTracerFactory.newServerStreamTracer(anyString(), any(Metadata.class)))
         .thenReturn(serverStreamTracer);
     callOptions = CallOptions.DEFAULT.withStreamTracerFactory(clientStreamTracerFactory);
+
+    doAnswer(
```

Comment:
Each test can do the same thing - store the messages it receives - but a few didn't need this, so I just loop over them (and don't close them, but correcting that). This can definitely be made nicer (using a concrete class instead of mocks, as you suggested in AbstractServerStreamTest), but I didn't want to invest too much additional time into redoing the tests until the API changes are approved/stabilized, as changes there could require completely different testing mechanism.

Incidentally, this change includes a couple fixes to the flow control pushback test that - in local testing - seem to deflake it. That's orthogonal to this, but hopefully a nice benefit once I invest the time to clean up the mocking/Answers used here.